cmake/modules/CheckCompiler.cmake

```diff
@@ -152,23 +152,6 @@ set(CMAKE_CXX_STANDARD ${CXX_STANDARD_STRING} CACHE STRING "")
 set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
 set(CMAKE_CXX_EXTENSIONS FALSE CACHE BOOL "")
 
-if(cxx11 OR cxx14 OR cxx17)
-  message(DEPRECATION "Options cxx11/14/17 are deprecated. Please use CMAKE_CXX_STANDARD instead.")
```

Comment:
What about keeping:
```
if(cxx11 OR cxx14 OR cxx17)
   message(FATAL "Options cxx11/14/17 are have removed.  Please use CMAKE_CXX_STANDARD instead.  C++17 or higher is required.")
endif()

Reply:
But when would we be allowed to drop that? IMHO 5 years of deprecation is really enough. Plus they don't serve any purpose anymore because we require building with C++17 now.

Reply:
You would 'never' drop those 2 lines.  The purpose is to stop a user that is rerunning their cmake script or continue to use their old build to have the C++17 turned on without them knowing.  I.e.  I propose to make the usage of those variables "Fatal" to obviously fail rather than silently do something different than the user expects.

Reply:
I agree with Philippe that erroring out and stopping the build if one of the options are used is a good approach. I wonder though, would removing those lines entirely make the build fail or does cmake silently ignore the option if it doesn't exist in the build system? If the second is the case, then we are just left with throwing a `Fatal`

Reply:
> cmake silently ignore the option if it doesn't exist in the build system? 

It silently ignore additional variable/options.

Reply:
FWIW I'm still not very happy with this approach because it will never allow us to fully drop configuration options and I don't think that's the right way. That said, I added it to the list of removed options in `cmake/modules/RootBuildOptions.cmake` that we appear to already maintain...

Reply:
>  it will never allow us to fully drop configuration options 

Besides saving a few characters, what would be the advantage of that?  (i.e. we can't reuse the options and 'arbitrary' options should not really have any effects on the build).

I still find it much clearer for the user to tell them that `cmake -Dcxx11=ON ...` is not building what they think they are building (that it is not possible to build what they asked for is another issue entirely :) ).

> That said, I added it to the list of removed options

Thanks for finding this, I had forgotten about and indeed is the right place for it.